R/utils.R

```diff
@@ -757,3 +757,67 @@ calc_sd <- function(x) {
   }
 }
 
+
+#' send email by wrapping `mailR::send.mail`
+#'
+#' send email by wrapping `mailR::send.mail`
+#'
+#' @param body a body content of the email
+#' @param to a string with the recipient
+#' @param subject a string with the subject
+#' @param from a string with the sender
+#' @param html a logical flag indicating whether the body of the email should be parsed as HTML
+#' @param inline a logical flag indicating whether images in the HTML file should be embedded inline
+#' @param host_name a string with the hostname
+#' @param attached_files a character with file paths to be attached to the email
+#'
+#' @keywords utils
+#' @export
+send_email <-
+  function(body,
+           subject,
+           to = get_env_var("EMAIL_RECIPIENT"),
+           from = get_env_var("EMAIL_SENDER"),
+           html = TRUE,
+           inline = FALSE,
+           host_name = get_env_var("EMAIL_HOSTNAME"),
+           attached_files = NULL) {
+
+    checkmate::assert_character(body)
+    checkmate::assert_string(subject, min.chars = 1)
+    checkmate::assert_string(to, min.chars = 1)
+    checkmate::assert_from(from, min.chars = 1)
+    checkmate::assert_flag(html)
+    checkmate::assert_flag(inline)
+    checkmate::assert_from(host_name, min.chars = 1)
+    checkmate::assert_character(attached_files)
```

Comment:
These two lines should be wrapped into if, to allow for using def arg with NULL, i.e.

```
    if (!is.null(attached_files)) {
        checkmate::assert_character(attached_files)
        checkmate::assert_file_exists(attached_files)
    }
```

Reply:
done in 708cae47e89427a20c692bfe2f47214d6761554f.